src/ui/pages/RevealPrivateKey/RevealPrivateKey.tsx

```diff
@@ -0,0 +1,171 @@
+import { useWalletGroup } from 'src/ui/shared/requests/useWalletGroups';
+import { usePrivateKey } from './usePrivateKey';
+
+function Reveal({
+  address,
+  onSubmit,
+  onSessionExpired,
+}: {
+  groupId: string;
+  address: string;
+  onSubmit: () => void;
+  onSessionExpired(): void;
+}) {
+  const { data: privateKey, isLoading, error } = usePrivateKey(address);
+  const { handleCopy, isSuccess: isCopySuccess } = useCopyToClipboard({
+    text: privateKey || '',
+  });
+  useEffect(() => {
+    if (isSessionExpiredError(error)) {
+      onSessionExpired();
+    }
+  }, [error, onSessionExpired]);
+  if (isLoading) {
+    return <ViewLoading />;
+  }
+  if (!privateKey) {
+    if (isSessionExpiredError(error)) {
+      return null;
+    }
+    throw new Error('Could not get private key');
+  }
+  return (
+    <PageColumn>
+      <PageTop />
+      <UIText kind="body/regular">
+        <span>
+          Your private key can be used to access all of your funds. Do not share
+          it with anyone
+        </span>
+      </UIText>
+      <Spacer height={24} />
+      <VStack gap={16}>
+        <BlurredToggle>
+          <Surface
+            padding={16}
+            style={{
+              paddingRight: 36, // because of toggle button on the right
+            }}
+          >
+            <UIText kind={'body/regular'} style={{ wordBreak: 'break-word' }}>
+              {privateKey}
+            </UIText>
+          </Surface>
+        </BlurredToggle>
+        <div style={{ textAlign: 'center' }}>
+          <Button
+            kind="regular"
+            size={36}
+            type="button"
+            onClick={handleCopy}
+            style={{ paddingLeft: 16, paddingRight: 16 }}
+          >
+            <HStack gap={8}>
+              {React.createElement(isCopySuccess ? CheckIcon : CopyIcon, {
+                style: { display: 'block', width: 20, height: 20 },
+              })}
+              <ZStack
+                hideLowerElements={true}
+                justifyContent="start"
+                style={{ textAlign: 'left' }}
+              >
+                {isCopySuccess ? <span>Copied to Clipboard</span> : null}
+                <span aria-hidden={isCopySuccess}>Copy to Clipboard</span>
+              </ZStack>
+            </HStack>
+          </Button>
+        </div>
+      </VStack>
+      <Button
+        style={{ marginTop: 'auto' }}
+        autoFocus={true}
+        onClick={() => onSubmit()}
+      >
+        Done
+      </Button>
+      <PageBottom />
+    </PageColumn>
+  );
+}
+
+export function RevealPrivateKey() {
+  useBackgroundKind(whiteBackgroundKind);
+
+  const [params, setSearchParams] = useSearchParams();
+  const groupId = params.get('groupId');
+  const address = params.get('address');
+  invariant(groupId, 'groupId param is required for RevealPrivateKey view');
+  invariant(address, 'address param is required for RevealPrivateKey view');
+
+  const updateSearchParam = (
```

Comment:
We've been using this helper a lot. May be it worth extracting into a file?

Reply:
Yep, I noticed it's duplicated in at least two places. I thought about fixing it, but I've already made so many changes in this PR. Since you mentioned it, I'll go ahead and extract it :)